paf-mvp-core-js/src/model/model-signature.service.ts

```diff
@@ -0,0 +1,63 @@
+import { IDSASigner } from '@core/crypto/digital-signature';
+import { IdsAndPreferences, Seed } from './generated-model';
+import { UnsignedSource } from './model';
+
+export const SIGN_SEP = '\u2063';
+
+/**
+ * Service for signing models
+ */
+export interface IModelSignatureService {
+  /**
+   * Sign a seed. The signature relies on the ids and preferences of the user.
+   */
+  signSeed(seed: UnsignedSource<Seed>, idsAndPreferences: IdsAndPreferences): Promise<Seed>;
+}
+
+/**
+ * Construct string used for generating the signature of entities.
+ */
+export interface ISignedStringBuilder {
+  /**
+   * Generate the string that must be signed for the Source of the a Seed.
+   */
+  buildStringForSigningSeed(seed: UnsignedSource<Seed>, idsAndPreferences: IdsAndPreferences): string;
+}
+
+export class ModelSignatureService implements IModelSignatureService {
+  constructor(
+    private readonly signer: IDSASigner,
+    private readonly stringBuilder: ISignedStringBuilder = new SignedStringBuilder()
+  ) {}
+
+  async signSeed(unsignedSeed: UnsignedSource<Seed>, idsAndPreferences: IdsAndPreferences): Promise<Seed> {
+    const signedString = this.stringBuilder.buildStringForSigningSeed(unsignedSeed, idsAndPreferences);
```

Comment:
I think the name `signedString` is not accurate. As mentioned below, maybe `inputString` or `signingString`, but not "signed".

Reply:
Sorry, I saw this comment after answering the previous one. `signedSigned` string is more accurate. `inputString` is too abstract and for `SigningString`, the string is not the signer it is the element that is signed.

Reply:
the string that this method return is not yet signed: I find it pretty confusing to name "signed" a variable that _will be signed_. Going this direction, the only option that makes sense to me would be `stringToSign` (which I find quite ugly, but whatever)